packages/@guardian/source-react-components/scripts/fetch-icons.ts

```diff
@@ -0,0 +1,257 @@
+			values: [
+				{ value: '0 0 24 24', newValue: '-3 -3 30 30' },
+				// legacy 50x20 wide viewbox
+				{ value: '0 0 48 24', newValue: '-1 2 50 20' },
+			],
+		},
+	]);
+
+	if (plugins.length === 0) {
+		return {};
+	}
+
+	babelConfig.plugins = plugins;
+
+	return babelConfig;
+};
+
+// now we start fetching SVGs and building our components
+void (async () => {
+	// remove any existing icons from a previous run
+	try {
+		await rm(vendorIconPath, { recursive: true });
+	} catch (e) {
+		// do nothing
+	}
+
+	// create the icons directory
+	await mkdirp(vendorIconPath);
+
+	// create a readme
+	await writeFile(path.resolve(vendorIconPath, 'README.md'), readme, 'utf8');
+
+	// Get a list of available (figma) components from Figma
+	// https://www.figma.com/developers/api#library-items-types
+	const figmaComponents = (
+		await axios.get<FigmaComponentsResponse>(
+			`https://api.figma.com/v1/files/${ICON_FILE}/components`,
+			{
+				headers: {
+					'X-Figma-Token': process.env.FIGMA_TOKEN ?? 'ADD ME!!!',
+				},
+			},
+		)
+	).data.meta.components;
+
+	// filter out the icons from the list of figma components
+	const figmaIconComponents = figmaComponents.filter((c) => {
+		return (
+			// Only get icons that are in a certain frame
+			c.containing_frame &&
+			c.containing_frame.name === 'UI icons 24(w)x24(w)'
+		);
+	});
+
+	const iconIds = figmaIconComponents.map(({ node_id }) => node_id).join(',');
+
+	// Get the URLs we can fetch actual images from from Figma
+	// https://www.figma.com/developers/api#get-images-endpoint
+	const figmaIconSvgUrlsByNodeId = (
+		await axios.get<FigmaImagesResponse>(
+			`https://api.figma.com/v1/images/${ICON_FILE}/?ids=${iconIds}&format=svg`,
+			FIGMA_API_OPTIONS,
+		)
+	).data.images;
+
+	// fetch the SVGs from Figma, create a react component from them using svgr
+	// and save them both
+	for (const icon of figmaIconComponents) {
+		const url = figmaIconSvgUrlsByNodeId[icon.node_id];
+
+		// Fetch SVG markup from Figma
+		const svg = (await axios.get<string>(url)).data;
+
+		// SVGR template
+		// https://react-svgr.com/docs/node-api/
+		const template: Config['template'] = (variables, { tpl }) => {
+			return tpl`
+			// DO NOT EDIT
+			// this file is auto-generated by scripts/fetch-icons.ts
+
+			import { css } from '@emotion/react';
+			import type { EmotionJSX } from '@emotion/react/types/jsx-namespace';
+			import { iconSize, visuallyHidden } from '@guardian/source-foundations';
+			import type { IconProps } from '../../src/@types/Icons';
+
+			${variables.imports};
+
+			const ${variables.componentName} = ({
+				size,
+			}: IconProps): EmotionJSX.Element => (
+				${variables.jsx}
+			);
+		  `;
+		};
+
+		const svgProps = {
+			focusable: '{false}',
+			'aria-hidden': '{true}',
+			width: WIDE_ICONS.includes(icon.name)
+				? '{undefined}'
+				: '{size ? iconSize[size] : undefined}',
+			height: WIDE_ICONS.includes(icon.name)
+				? '{size ? iconSize[size] : undefined}'
+				: '{undefined}',
```

Comment:
Worth noting that if CSS fails to load, SVG elements will default to 100% width, and their aspect ratio is defined by the `viewBox`. This might not be an issue in practice as I think we inject CSS directly into the HTML, but can otherwise cause very large icons taking over the whole viewport.